### ISSUE TITLE
Passing arguments to the unicorn view

### ISSUE THREAD

**Issue by nyradr** [adamghill/django-unicorn#69]
Being able to send arguments from the parent django template to the unicorn view.

To be able to do something like :

```python
class TestView(UnicornView):
    name = "hello-world"
    def __init__(self, var):
         self.var = var
    def set_var(self, var):
         self.var = var
```
```j2
{% unicorn 'hello-world' with var=var %}      (django {% include %} like )
```

Or to be able to send arguments to Unicorn.call :
```j2
{% unicorn 'hello-world' %}

<button onclick="Unicorn.call('hello-world', 'set_var', 'var_value');">Set the name from outside the component</button>
```

Even if it's just sending a one-shot primitives it will be very helpful to give some context to the unicorn view.

**Comment by adamghill** [comment]
This makes sense and I think it should be relatively straight-forward to do. I'll try to get it into the next release. 👍

**Comment by adamghill** [comment]
@nyradr https://github.com/adamghill/django-unicorn/pull/70 has some initial work for this issue. Currently, the API would look like:

```python
class TodoView(UnicornView):
    hello = "initial"

    def __init__(self, *args, **kwargs):
        super().__init__(**kwargs)
        self.hello = kwargs.get("hello", "default")
```

```
{% unicorn 'todo' hello='world' %}
```

Even this would work (assumes that `test` is a view context variable with a field of `variable` (e.g. something like `{ "variable": "hello" }).

```
{% unicorn 'todo' hello=test.variable %}
```

Does that look like a reasonable API? I initially made it work with kwargs _and_ regular args that could passed in, but supporting the args made the components a little more brittle/complicated and harder to just add new arguments in my opinion.

I could also require `with` in the template tag to signal that the rest of the tag is kwargs (i.e. `{% unicorn 'todo' with hello='world' %}`, but not sure if that's necessary.

Any thoughts?

**Comment by adamghill** [comment]
>send arguments to Unicorn.call 

I _think_ this already works as expected (although tell me if you find an example that doesn't seem to work).

For this component:

```python
class HelloWorldView(UnicornView):
    name = "World"

    def set_name(self, name):
        self.name = name
```

This template should work:

```
<button onclick='Unicorn.call("hello_world", "set_name(\"hello\")");'>Method call from outside of the component</button>
```

The shortcut of setting the field directly should also work:

```
<button onclick='Unicorn.call("hello_world", "name=\"hello\"");'>Method call from outside of the component</button>
```

And either way should work with Django template variables if it is in the current context:

```
<button onclick='Unicorn.call("hello_world", "name=\"{{ context.variable }}\"");'>Method call from outside of the component</button>
```

It's a little messy because of the `\"` for the strings, but I can't think of an easy way to get around that issue.

Let me know if that doesn't make sense or doesn't work for your use case!

**Comment by nyradr** [comment]
I managed to make it work that way (making the user click on a button to initialize a component is not very user friendly :) ) :

```j2
{% unicorn 'trade-add' %}
 <script>
    function f() {
        Unicorn.call("hello-world", "set_name({{ object.id }})");
    }
    window.onload = f;
</script>
```
But it's not that practical and a bit ugly.
A simple one-liner line `{% unicorn 'hello-world' hello=object.id %}` (handling the variables in an __init__ function is not a problem) would be perfect.

**Comment by adamghill** [comment]
The PR has been merged and tagged with 0.10.0 which just got released to pypi. I'll update the documentation site soon as well. But, feel free to try it out and let me know if you run into any problems.

**Comment by adamghill** [comment]
Documentation is now live at https://www.django-unicorn.com/documentation/components#component-arguments.
